### PR TITLE
Gitignore update & cleaning code

### PR DESCRIPTION
**Pull request by otradovec**
Unnecessary comments just spoil the code.

### CODE REVIEW

**Comment by hbitteur** [comment]
OK for adding new elements to .gitignore file as needed.

Not OK for removing comments from source file.
Granted, this little file (AudiverisTest.java) is not at all representative of typical Audiveris files, so I don't really care for this file. By the way, it no longer tests what it was meant to test: subsequent calls of Audiveris main entry. Perhaps we should resurrect this test file.

As far as comments are concerned, I insist in keeping (or reinserting) the comment lines that are used as standard visual separators between code sections in a Java file, like the following:
```java
public class FooBar {

    //~ Static fields/initializers -----------------------------------------------------------------

    //~ Instance fields ----------------------------------------------------------------------------

    //~ Constructors -------------------------------------------------------------------------------

    //~ Methods ------------------------------------------------------------------------------------

    //~ Inner Classes ------------------------------------------------------------------------------

}
```

I removed them all about one+ year ago. It was a mistake. I will reinsert them before the next release is out.

**Comment by otradovec** [comment]
"I don't really care for this file" will You merge it then?
"it no longer tests what it was meant to test" Does this file test something, if not could be deleted, right?
"I insist in keeping (or reinserting) the comment lines that are used as standard visual separators" which standard do you mean? Never heard of any standard like that, in fact according to principles of professional & clean code Robert C. Martin writes in his book Clean code in section Bad comments, subsection Position markers: 
"Sometimes programmers like to mark a particular position in a source file. For example, I recently found this in a program I was looking through:// Actions //////////////////////////////////
There  are  rare  times  when  it  makes  sense  to  gather  certain  functions  together  beneath  a banner like this. But in general they are clutter that should be eliminated—especially the noisy train of slashes at the end."
And I thing this is classic example, there is no need to state that there comes constructors and then methods, everyone sees that from the code. What is the purpose of an empty constructor in that test class anyway?

**Comment by hbitteur** [comment]
@otradovec 
I thought you were trying to improve some of the Audiveris unitary test files. Hence my remark about the AudiverisTest.java file which, if I remember correctly, was initially meant to test repetitive calls to Audiveris main entry point. For some reason I don't remember, I obviously had commented out the body of this test. When I proposed you to resurrect it, I meant to re-test this feature of multiple calls.

Now, looking at the additional files in this second pull request, I can't detect any concrete modification except the removal of blank lines, comments and constructors. What are you trying to achieve? This is not clear for me.

I won't argue for a few unitary test files, they represent just 8_500 lines and are just that: small unitary tests for which logic is very basic. But if your goal is to extend these modifications to Audiveris main source code (about 280_000 lines as of this writing), please don't do it.

Code presentation is a matter of personal taste I agree, but it's something important, it eases code reading and mastering.
In the past, I used to apply Jalopy pretty formatter on each Java file. The "standard" separators I mentioned were inserted by  that Jalopy tool, so even though they do not pretend to be "ISO standards", my opinion is they are rather well chosen.
I cannot use Jalopy anymore (the free version does not accept any "<>" diamond, not to speak of more recent lambda expressions).
Today I use the source pretty formatter included in NetBeans IDE, it is less powerful than Jalopy but it accepts even the most recent Java features. I also tried Eclipse but was not really convinced.

I noticed you modified the .gitignore file to add something related to IntelliJ I think. If you are familiar with this IDE, could you tell us if they provide a good Java pretty formatter? If so, I'm interested.

**Comment by otradovec** [comment]
@hbitteur 
Firstly I thought about improving production code, because I think, there is a lot of wasted potential. I am very convinced that some small improvements would change this app from useless to killer app. But then I realised that production code is too messy to give me any idea about how this works, so I turned to test but realised the same thing. So I decided to first clean everything before making any improvement.
You are right that I use Idea IDE, I have tried Netbeans and Eclipse also, plus Qt, Atom and VS. Idea is clear winner for me.
image
If I look at the code, in my eyes it is clear that there is just too many useless comments. The real code is basically lost between the comments. What do you say about me deleting all  
`    //--------//
    // getCli //
    //--------//`?

**Comment by otradovec** [comment]
And what should happen when there is the second call? Should be the second call ignored?

**Comment by otradovec** [comment]
I have found out that the options mentioned in the test does no exist. The batch is legit, but the -step does not support EXPORT so I changed it to LOAD because I see no reason for any other if we just test loading concurrent loading of the app. Also the option "-input" does not exist. What should be placed instead?

**Comment by hbitteur** [comment]
> If I look at the code, in my eyes it is clear that there is just too many useless comments. The real code is basically lost between the comments. What do you say about me deleting all
//--------// // getCli // //--------//?

I clearly say: don't do it. 
These are not useless comments, but visual markers to ease the reading of code structure. 
I remember someone saying that one line of code is typically read 5 times more often that it is written.
Hence the concern about code visual reading. Similarly, many IDE today provide syntax highlighting to ease such reading.

Please don't think you would improve Audiveris code by removing everything you don't like, be it comment, constructor, line breaks, etc.
I put them in the code on purpose. Less lines does not mean better code.

**Comment by hbitteur** [comment]
> Firstly I thought about improving production code, because I think, there is a lot of wasted potential. I am very convinced that some small improvements would change this app from useless to killer app. But then I realised that production code is too messy to give me any idea about how this works, so I turned to test but realised the same thing. So I decided to first clean everything before making any improvement.

Frankly, if you think Audiveris is useless, why would you spend any time with it? Just ignore it!

**Comment by hbitteur** [comment]
> I have found out that the options mentioned in the test does no exist. The batch is legit, but the -step does not support EXPORT so I changed it to LOAD because I see no reason for any other if we just test loading concurrent loading of the app. Also the option "-input" does not exist. What should be placed instead?

> And what should happen when there is the second call? Should be the second call ignored?

As a very basic rule, discussions / issues should stick to one topic at a time.
Here, you are mixing general opinions about how the code should look like with a precise issue on a old test file.

So my clear proposal from now on is as follows:
- I will not merge any pull request focused on "code de-formatting". You can instead use Audiveris new [Discussions forum](https://github.com/Audiveris/audiveris/discussions) for such topics.
- Any precise issue must be addressed in a separate dedicated issue / pull request.

**Comment by maximumspatium** [comment]
@otradovec As someone who works with Hervé on his brilliant application since 2008, I strongly disagree with the claim Audiveris' production code be too messy. The fact that you don't understand how it works lays in the nature of the complex algorithms involved. I spent significant amount of time to grasp what's going on there. I'm sure deleting comments won't improve that. As opposite, reading of academic papers will do. I can provide you with some useful links if you want.

Regarding your PR, I understand your decision to pick the low-hanging fruit. But before you criticize Audiveris' source, especially its coding style, you get to know first that some parts of it are more than 20 years old (!) - Hervé has started this application back in 1998.
BTW, the comments you want to remove have been automatically generated by an old IDE (I think it was [Netbeans](https://en.wikipedia.org/wiki/NetBeans)) so you have to teach its community to do professional coding, not us :wink:

TLDR; I'm very sorry but we won't take your patch because:
- it brings no improvements 
- mixes unrelated changes (one of the major PR guidelines says: **Don't submit unrelated changes in the same pull request**)
- affects old sources that will be likely removed soon

If you are serious about improving Audiveris, please join our Discord server at https://discord.gg/ccH3XcUS so we can discuss together what to do next.
We have a ton of plans and need professional coders. Don't hesitate to ask if you don't understand something. I believe it's not at all embarrassing to be confused by non-straightforward implementations of complex heavy-math algorithms.
